### PR TITLE
git-machete 3.12.0 (new formula)

### PR DESCRIPTION
**Pull request by amalota**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Hi Homebrew team,

This is the second approach to adding our software tool to Homebrew, our first try was 3 years ago: [#48238](https://github.com/Homebrew/homebrew-core/pull/48238). Since then, we grew a lot in both download stats, visibility (we recently got mentioned in an independently-written blog post https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/ ) and we also have got 500+ stars on [GitHub](https://github.com/VirtusLab/git-machete).

### CODE REVIEW

File: Formula/git-machete.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class GitMachete < Formula
+  include Language::Python::Virtualenv
+  desc "Git repository organizer & rebase workflow automation tool"
+  homepage "https://github.com/VirtusLab/git-machete"
+  url "https://pypi.org/packages/source/g/git-machete/git-machete-3.12.0.tar.gz"
+  sha256 "b999ff0ead4b856436ee74aae394ae3c68941fab8286256488304382d0fc8452"
+  license "MIT"
+  depends_on "python@3.10"
```
Comment (carlocab) [inline Formula/git-machete.rb:8]:
```suggestion
  license "MIT"

  depends_on "python@3.10"
```

File: Formula/git-machete.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class GitMachete < Formula
+  include Language::Python::Virtualenv
+  desc "Git repository organizer & rebase workflow automation tool"
```
Comment (carlocab) [inline Formula/git-machete.rb:3]:
```suggestion
  include Language::Python::Virtualenv

  desc "Git repository organizer & rebase workflow automation tool"
```

**Review by carlocab — COMMENTED** [review]
Just some minor style suggestions.

When you make these changes, please do `git rebase` to get rid of the merge commit and squash your commits together. Thanks!

**Comment by PawelLipski** [comment]
@carlocab if it comes to maintenance of this formula... is there some mechanism in place in homebrew-core to automatically update the `url` and `sha256` fields once a new version is released to PyPI? or should we cook up some custom update mechanism (as we're doing with out tap currently: https://github.com/VirtusLab/homebrew-git-machete/commits/master)?

**Comment by carlocab** [comment]
You can just run
```
brew bump-formula-pr --version=$version git-machete
```
or use this action: https://github.com/dawidd6/action-homebrew-bump-formula

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**